jlm/mlir/backend/JlmToMlirConverter.hpp

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2023 Magnus Sjalander <work@sjalander.com>
+ * See COPYING for terms of redistribution.
+ */
+
+#ifndef JLM_MLIR_BACKEND_JLMTOMLIRCONVERTER_HPP
+#define JLM_MLIR_BACKEND_JLMTOMLIRCONVERTER_HPP
+
+// JLM
+#include <jlm/llvm/ir/operators/lambda.hpp>
+#include <jlm/llvm/ir/RvsdgModule.hpp>
+
+// MLIR RVSDG dialects
+#include <JLM/JLMDialect.h>
+#include <RVSDG/RVSDGDialect.h>
+#include <RVSDG/RVSDGPasses.h>
+
+// MLIR generic dialects
+#include <mlir/Dialect/Arith/IR/Arith.h>
+
+namespace jlm::mlir
+{
+
+class JlmToMlirConverter final
+{
+public:
+  JlmToMlirConverter()
+  {
+    Context_ = std::make_unique<::mlir::MLIRContext>();
+    // Load the RVSDG dialect
+    Context_->getOrLoadDialect<::mlir::rvsdg::RVSDGDialect>();
+    // Load the JLM dialect
+    Context_->getOrLoadDialect<::mlir::jlm::JLMDialect>();
+    // Load the Arith dialect
+    Context_->getOrLoadDialect<::mlir::arith::ArithDialect>();
+    Builder_ = std::make_unique<::mlir::OpBuilder>(Context_.get());
```

Comment:
If the dialects do not need to be loaded before the OpBuilder is created, then you could also add the initialization of Builder_ into the constructor list.

Reply:
I've not found a way of moving it to the constructor list